stream-rpc/src/connection/manager.rs

```diff
@@ -0,0 +1,129 @@
+// Copyright (c) The Diem Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::{
+    connection::{BoxConnectionStream, Client, ConnectionContext, StreamSender},
+    subscriptions::SubscriptionConfig,
+    counters, logging,
+};
+use diem_logger::debug;
+use futures::StreamExt;
+use std::{
+    collections::HashMap,
+    sync::{
+        atomic::{AtomicU64, Ordering},
+        Arc, RwLock,
+    },
+};
+use storage_interface::DbReader;
+
+/// Our global unique user id counter.
+static NEXT_USER_ID: AtomicU64 = AtomicU64::new(0);
+
+#[derive(Clone)]
+pub struct ConnectionManager {
+    pub clients: Arc<RwLock<HashMap<u64, Client>>>,
+    pub diem_db: Arc<dyn DbReader>,
+    pub config: Arc<SubscriptionConfig>,
+}
+
+impl ConnectionManager {
+    pub fn new(diem_db: Arc<dyn DbReader>, config: Arc<SubscriptionConfig>) -> Self {
+        Self {
+            clients: Arc::new(RwLock::new(HashMap::new())),
+            diem_db,
+            config,
+        }
+    }
+
+    fn get_db(&self) -> Arc<dyn DbReader> {
+        self.diem_db.clone()
+    }
+
+    pub fn has_client(&self, client_id: &u64) -> bool {
+        self.clients.read().unwrap().contains_key(client_id)
+    }
```

Comment:
you should take in client_id by value, not reference since client_id is `Copy`

Reply:
does it matter that it's going to be passed by reference into `contains_key` right after?

Reply:
nope, this is just the more idiomatic thing to do